lib/optimization/event.go

```diff
@@ -181,15 +178,15 @@ func (t *TableData) Rows() uint {
 	return uint(len(t.rowsData))
 }
 
-func (t *TableData) DistinctDates(ctx context.Context, colName string) ([]string, error) {
+func (t *TableData) DistinctDates(additionalDateFmts []string, colName string) ([]string, error) {
 	retMap := make(map[string]bool)
 	for _, row := range t.rowsData {
 		val, isOk := row[colName]
 		if !isOk {
 			return nil, fmt.Errorf("col: %v does not exist on row: %v", colName, row)
 		}
 
-		extTime, err := ext.ParseFromInterface(ctx, val)
+		extTime, err := ext.ParseFromInterface(additionalDateFmts, val)
```

Comment:
Seems like `additionalDateFmts` should be the second argument since `val` is what's being parsed.